File: in_java/Pattern_11.java
package in_java;

public class Pattern_11 extends Pattern {
    public Pattern_11(int row, int column, String symbol) {
        super(row, column, symbol);
    }

    public void print() {
        int halfRow = this.row / 2;
        int tempI;
        for (int i = 0; i < this.row; i++) {
            for (int j = 0; j < this.column; j++) {
                if (i <= halfRow) {
                    if (j <= i) {
                        System.out.print(this.symbol);
                    } else {
                        System.out.print(" ");
                    }
                } else {
                    tempI = row - 1 - i;
                    if (j <= tempI) {
                        System.out.print(this.symbol);
                    } else {
                        System.out.print(" ");
                    }
                }
            }
            System.out.println();
        }
    }
}

File: in_typescript/src/Pattern_11.ts
export default function pattern_11() {
  let row = 9;
  let column = 5;

  let symbol = "*";
  let halfRow = Math.floor(row / 2);
  let tempI = 0;
  for (let i = 0; i < row; i++) {
    for (let j = 0; j < column; j++) {
      if (i <= halfRow) {
        if (j <= i) {
          process.stdout.write(`${symbol}`);
        } else {
          process.stdout.write(" ");
        }
      } else {
        tempI = row - 1 - i;
        if (j <= tempI) {
          process.stdout.write(`${symbol}`);
        } else {
          process.stdout.write(" ");
        }
      }
    }
    process.stdout.write("\n");
  }
}
